zmon_aws_agent/main.py

```diff
@@ -96,161 +128,168 @@ def main():
-        scaling_groups = aws.get_auto_scaling_groups(region, infrastructure_account)
-        rds = aws.get_rds_instances(region, infrastructure_account, entities)
-        elasticaches = aws.get_elasticache_nodes(region, infrastructure_account)
-        dynamodbs = aws.get_dynamodb_tables(region, infrastructure_account)
-        certificates = aws.get_certificates(region, infrastructure_account)
-        aws_limits = aws.get_limits(region, infrastructure_account, apps, elbs, entities)
-        sqs = aws.get_sqs_queues(region, infrastructure_account, entities)
-        postgresql_clusters = postgresql.get_postgresql_clusters(region, infrastructure_account,
-                                                                 scaling_groups, apps)
-
-    account_alias = aws.get_account_alias(region)
-    ia_entity = {
-        'type': 'local',
-        'infrastructure_account': infrastructure_account,
-        'account_alias': account_alias,
-        'region': region,
-        'id': 'aws-ac[{}:{}]'.format(infrastructure_account, region),
-        'created_by': 'agent',
-    }
-
-    application_entities = aws.get_apps_from_entities(apps, infrastructure_account, region)
-
-    if args.postgresql_user and args.postgresql_pass:
-        postgresql_clusters = zmon_client.get_entities({
+    init_opentracing_tracer(args.tracer)
+    root_span = opentracing.tracer.start_span(operation_name='aws_entity_discovery')
+    with root_span:
+
+        logging.basicConfig(level=logging.INFO)
+        # 0. Fetch extra data for entities
+        entity_extras = {}
+        for ex in os.getenv('EXTRA_ENTITY_FIELDS', '').split(','):
+            if '=' not in ex:
+                continue
+            k, v = ex.split('=', 1)
+            if k and v:
+                entity_extras[k] = v
+
+        # 1. Determine region
+        if not args.region:
+            logger.info('Trying to figure out region..')
+            try:
+                response = requests.get('http://169.254.169.254/latest/meta-data/placement/availability-zone',
+                                        timeout=2)
+            except Exception:
+                logger.exception('Region was not specified as a parameter and' +
+                                 'can not be fetched from instance meta-data!')
+                raise
+            region = response.text[:-1]
+        else:
+            region = args.region
+
+        root_span.set_tag('region', region)
+
+        logger.info('Using region: {}'.format(region))
+
+        logger.info('Entity service URL: %s', args.entityservice)
+
+        logger.info('Reading DNS data for hosted zones')
+        aws.populate_dns_data()
+
+        aws_account_id = aws.get_account_id(region)
+        infrastructure_account = 'aws:{}'.format(aws_account_id) if aws_account_id else None
+
+        if not infrastructure_account:
+            logger.error('AWS agent: Cannot determine infrastructure account ID. Terminating!')
```

Comment:
May be it makes sense to go over all the exceptions and log.error in the main and set ``error`` tag and ``exception`` log to the root span, would be easy in searching.